File: src/main/java/havarunner/exception/CodingConventionException.java
package havarunner.exception;

import havarunner.HavaRunner;

public class CodingConventionException extends RuntimeException {
    CodingConventionException(String messageAboutParticularCodingConventionViolation) {
        super(
            String.format(
                "%s (%s is strict about coding conventions, because its authors believe they help writing better software.)",
                messageAboutParticularCodingConventionViolation, HavaRunner.class.getSimpleName()
            )
        );
    }
}

File: src/main/scala/havarunner/exception/CodingConventionException.scala
package havarunner.exception

import havarunner.HavaRunner

class CodingConventionException(msg: String) extends RuntimeException(
  String.format(
    "%s (%s is strict about coding conventions, because its authors believe they help writing better software.)",
    msg, classOf[HavaRunner].getSimpleName
  )
)